docker-compose-net/docker-compose-net.md

```diff
@@ -0,0 +1,234 @@
+ is automatically disabled and is not used to attach containers. See [Integrating Docker via the Network Plugin](https://www.weave.works/docs/net/latest/features/#plugin).
+
+
+##<a name="setup-proxy"></a>Docker Compose and the Weave Net Docker API Proxy
+
+
+The following is a simple setup that shows how to configure docker-compose for Weave Net across two hosts using the Docker API proxy. You will launch the alpine using Docker Compose and then run 'ping' to test the connectivity between the two hosts. 
+
+**1.** Create two hosts on Virtualbox with docker machine in two different terminal windows: 
+
+~~~bash
+docker-machine create -d virtualbox weave-compose-01
+~~~
+
+And, in a new terminal window, create the other VM: 
+
+~~~bash
+docker-machine create -d virtualbox weave-compose-02
+~~~
+
+**2.** In order to peer hosts on Weave Net, the IP address of one of the VMs you just created must be known beforehand: 
+
+~~~bash
+docker-machine ip weave-compose-02
+~~~
+
+Make a note of the IP address that appears. 
+
+**3.** Install and then launch Weave Net onto both VMs. From one terminal windown launch Weave: 
+
+~~~bash
+sudo curl -L git.io/weave -o /usr/local/bin/weave
+sudo chmod +x /usr/local/bin/weave
+weave launch
+~~~
+
+And in a second terminal launch Weave onto the other host, and pass the IP of the first host to it: 
+
+~~~bash
+sudo curl -L git.io/weave -o /usr/local/bin/weave
+sudo chmod +x /usr/local/bin/weave
+weave launch <ip address host-02>
+~~~
+
+**4.** Setup the environment to run the Weave Docker API proxy: 
+
+~~~bash
+eval $(weave env)
+~~~
+
+With the Weave environment set up, you can run regular Docker commands on the same command line from which you created VMs. If you are attaching containers to the Weave Network, this step is mandatory, unless you've opted to use the Weave Net Docker Plugin which is described below. 
+
+**5.** Run the Docker Compose file and direct each service to a host: 
+
+~~~bash
+cd proxy-compose
+docker-compose up -d pingme
+~~~
+
+Next, switch to the second terminal, and change to the compose-plugin sub-directory and run the compose file:
+
+~~~bash
+cd proxy-compose
+docker-compose up -d pinger 
+~~~
+
+6. If all has went well, you should see 'Hello, Weave!' printed on your command line. Pingme will keep resonding to the command sent by Pinger for 3600 seconds. 
+
+
+
+###<a name="compose-proxy"></a>Docker Compose File Syntax for the Weave Docker API Proxy###
+
+The network_mode is set to "bridge" for this mode. 
+
+~~~bash
+version: '2'
+
+services:
+ pingme:
+    image: alpine
+    container_name: pingme
+    command: sleep 3600
+    command: nc -p 4000 -ll -e echo 'Hello, Weave!'
+    network_mode: "bridge"
+    
+ pinger:
+    image: alpine
+    container_name: pinger
+    command: ping pingme.weave.local
+    hostname: hello.weave.local
+    network_mode: "bridge"
+~~~
+
+
+##<a name="setup-plugin"></a>Docker Compose and the Weave Net Docker Plugin
+
```

Comment:
How is service discovery working in this scenario? Does it use Docker DNS? If so, we should tell the user that. Also, it's not clear that it would work as configured, unless Docker Swarm Mode is used.